ql/extractor/src/bin/generator.rs

```diff
@@ -1,542 +1,6 @@
-            dbscheme::Column {
-                unique: true,
-                db_type: dbscheme::DbColumnType::Int,
-                name: "child",
-                ql_type: ql::Type::At("container"),
-                ql_type_is_ref: true,
-            },
-        ],
-        keysets: None,
-    })
-}
-
-fn create_source_location_prefix_table<'a>() -> dbscheme::Entry<'a> {
-    dbscheme::Entry::Table(dbscheme::Table {
-        name: "sourceLocationPrefix",
-        keysets: None,
-        columns: vec![dbscheme::Column {
-            unique: false,
-            db_type: dbscheme::DbColumnType::String,
-            name: "prefix",
-            ql_type: ql::Type::String,
-            ql_type_is_ref: true,
-        }],
-    })
-}
-
-fn create_diagnostics<'a>() -> (dbscheme::Case<'a>, dbscheme::Table<'a>) {
-    let table = dbscheme::Table {
-        name: "diagnostics",
-        keysets: None,
-        columns: vec![
-            dbscheme::Column {
-                unique: true,
-                db_type: dbscheme::DbColumnType::Int,
-                name: "id",
-                ql_type: ql::Type::At("diagnostic"),
-                ql_type_is_ref: false,
-            },
-            dbscheme::Column {
-                unique: false,
-                db_type: dbscheme::DbColumnType::Int,
-                name: "severity",
-                ql_type: ql::Type::Int,
-                ql_type_is_ref: true,
-            },
-            dbscheme::Column {
-                unique: false,
-                db_type: dbscheme::DbColumnType::String,
-                name: "error_tag",
-                ql_type: ql::Type::String,
-                ql_type_is_ref: true,
-            },
-            dbscheme::Column {
-                unique: false,
-                db_type: dbscheme::DbColumnType::String,
-                name: "error_message",
-                ql_type: ql::Type::String,
-                ql_type_is_ref: true,
-            },
-            dbscheme::Column {
-                unique: false,
-                db_type: dbscheme::DbColumnType::String,
-                name: "full_error_message",
-                ql_type: ql::Type::String,
-                ql_type_is_ref: true,
-            },
-            dbscheme::Column {
-                unique: false,
-                db_type: dbscheme::DbColumnType::Int,
-                name: "location",
-                ql_type: ql::Type::At("location_default"),
-                ql_type_is_ref: true,
-            },
-        ],
-    };
-    let severities: Vec<(usize, &str)> = vec![
-        (10, "diagnostic_debug"),
-        (20, "diagnostic_info"),
-        (30, "diagnostic_warning"),
-        (40, "diagnostic_error"),
-    ];
-    let case = dbscheme::Case {
-        name: "diagnostic",
-        column: "severity",
-        branches: severities,
-    };
-    (case, table)
-}
+use codeql_extractor::generator::{generate, language::Language};
 
 fn main() -> std::io::Result<()> {
     tracing_subscriber::fmt()
```

Comment:
I guess we should also factor out all the command line argument and logger setup code. Feel free to postpone that to a follow-up PR though.